jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java

```diff
@@ -332,9 +332,20 @@ private static void deployPersistenceUnit(
         try {
             ValidatorFactory validatorFactory = null;
             final HashMap<String, ValidatorFactory> properties = new HashMap<>();
+
+            CapabilityServiceSupport css = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
             if (!ValidationMode.NONE.equals(pu.getValidationMode())) {
-                // Get the CDI-enabled ValidatorFactory
-                validatorFactory = deploymentUnit.getAttachment(BeanValidationAttachments.VALIDATOR_FACTORY);
+                if (css.hasCapability("org.wildfly.bean-validation")) {
+                    // Get the CDI-enabled ValidatorFactory
+                    validatorFactory = deploymentUnit.getAttachment(BeanValidationAttachments.VALIDATOR_FACTORY);
+                }
+                if (validatorFactory == null && ValidationMode.CALLBACK.equals(pu.getValidationMode())) {
```

Comment:
remove unneeded null check and comment, see below comment about how validation factory is only passed in to persistence provider if its available in the environment.